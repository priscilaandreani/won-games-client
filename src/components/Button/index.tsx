import { ReactNode } from 'react';
import * as S from './styles';

export type ButtonProps = {
  children?: ReactNode;
  $size?: 'small' | 'medium' | 'large';
  $fullWidth?: boolean;
  $icon?: ReactNode;
  onClick?: () => (event: React.MouseEvent<HTMLButtonElement>) => void;
};
const Button = ({
  children,
  $size = 'medium',
  $fullWidth = false,
  $icon,
  ...props
}: ButtonProps) => (
  <S.Wrapper
    $size={$size}
    $fullWidth={$fullWidth}
    $hasIcon={!!$icon}
    {...props}
  >
    {$icon}
    {!!children && <span>{children}</span>}
  </S.Wrapper>
);

export default Button;
