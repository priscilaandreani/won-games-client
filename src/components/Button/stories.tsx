import { Meta, StoryObj } from '@storybook/react';
import Button from '.';
import { AddShoppingCart } from '@styled-icons/material-outlined/AddShoppingCart';
import { ReactNode } from 'react';

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['small', 'medium', 'large']
      }
    },
    children: {
      type: 'string'
    },
    icon: {
      type: '' as ReactNode
    }
  }
} as Meta;

export const Default: StoryObj = {
  args: {
    children: 'Buy now!'
  }
};

export const withIcon: StoryObj = {
  args: {
    children: 'Buy now!',
    size: 'small',
    icon: <AddShoppingCart />
  }
};
