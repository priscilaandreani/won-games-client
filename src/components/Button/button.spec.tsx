import { screen } from '@testing-library/react';

import Button from '.';
import { renderWithTheme } from '@/utils/tests/helpers';

describe('<Button />', () => {
  it('should render the medium button by default', () => {
    renderWithTheme(<Button>Click me!</Button>);
    expect(screen.getByRole('button', { name: /click me!/i })).toHaveStyle({
      padding: '0.8rem 3.2rem',
      height: '4rem',
      fontSize: '1.4rem'
    });
  });

  it('should render the small size', () => {
    renderWithTheme(<Button size="small">Click me!</Button>);
    expect(screen.getByRole('button', { name: /click me!/i })).toHaveStyle({
      height: '3rem',
      fontSize: '1.2rem'
    });
  });

  it('should render the large size', () => {
    renderWithTheme(<Button size="large">Click me!</Button>);
    expect(screen.getByRole('button', { name: /click me!/i })).toHaveStyle({
      height: '5rem',
      fontSize: '1.6rem',
      padding: '0.8rem 4.0rem'
    });
  });
});
