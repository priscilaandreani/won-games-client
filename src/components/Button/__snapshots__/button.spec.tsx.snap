// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render the medium button by default 1`] = `
<div>
  <button
    class="sc-braxZu hhAYUC"
  >
    <span>
      Click me!
    </span>
  </button>
</div>
`;
